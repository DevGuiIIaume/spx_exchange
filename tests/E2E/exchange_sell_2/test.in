2
[T0] SELL 0 GPU 10 30;
[T0] SELL 1 GPU 20 30;
[T0] SELL 2 Router 50 35;
[T0] SELL 3 GPU 30 30;
[T0] SELL 4 Router 50 35;
[T1] SELL 0 GPU 15 30;
[T1] SELL 1 Router 55 30;
[T1] SELL 2 GPU 25 30;
[T1] SELL 3 Router 50 35;
[T0] DISCONNECT;
[T1] DISCONNECT;
