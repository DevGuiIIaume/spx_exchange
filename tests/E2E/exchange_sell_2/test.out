[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Created FIFO /tmp/spx_exchange_1
[SPX] Created FIFO /tmp/spx_trader_1
[SPX] Connected to /tmp/spx_exchange_1
[SPX] Connected to /tmp/spx_trader_1
[SPX] [T0] Parsing command: <SELL 0 GPU 10 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 10 @ $30 (1 order)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <SELL 1 GPU 20 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 30 @ $30 (2 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <SELL 2 Router 50 35>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 30 @ $30 (2 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 50 @ $35 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <SELL 3 GPU 30 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 60 @ $30 (3 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 50 @ $35 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <SELL 4 Router 50 35>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 60 @ $30 (3 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 100 @ $35 (2 orders)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 0 GPU 15 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 75 @ $30 (4 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 100 @ $35 (2 orders)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 1 Router 55 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 75 @ $30 (4 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 2
[SPX]		SELL 100 @ $35 (2 orders)
[SPX]		SELL 55 @ $30 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 2 GPU 25 30>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 100 @ $30 (5 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 2
[SPX]		SELL 100 @ $35 (2 orders)
[SPX]		SELL 55 @ $30 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 3 Router 50 35>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 1
[SPX]		SELL 100 @ $30 (5 orders)
[SPX]	Product: Router; Buy levels: 0; Sell levels: 2
[SPX]		SELL 150 @ $35 (3 orders)
[SPX]		SELL 55 @ $30 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] Trader 0 disconnected
[SPX] Trader 1 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $0
